configuration/redis/redis.go

```diff
@@ -30,35 +28,38 @@ import (
 
 	"github.com/dapr/components-contrib/configuration"
 	"github.com/dapr/components-contrib/configuration/redis/internal"
+	rediscomponent "github.com/dapr/components-contrib/internal/component/redis"
 	contribMetadata "github.com/dapr/components-contrib/metadata"
 	"github.com/dapr/kit/logger"
 )
 
 const (
-	connectedSlavesReplicas   = "connected_slaves:"
-	infoReplicationDelimiter  = "\r\n"
-	host                      = "redisHost"
-	password                  = "redisPassword"
-	enableTLS                 = "enableTLS"
-	maxRetries                = "maxRetries"
-	maxRetryBackoff           = "maxRetryBackoff"
-	failover                  = "failover"
-	sentinelMasterName        = "sentinelMasterName"
-	redisDB                   = "redisDB"
-	defaultBase               = 10
-	defaultBitSize            = 0
-	defaultDB                 = 0
-	defaultMaxRetries         = 3
-	defaultMaxRetryBackoff    = time.Second * 2
-	defaultEnableTLS          = false
-	redisWrongTypeIdentifyStr = "WRONGTYPE"
+	connectedSlavesReplicas      = "connected_slaves:"
+	infoReplicationDelimiter     = "\r\n"
+	host                         = "redisHost"
+	password                     = "redisPassword"
+	enableTLS                    = "enableTLS"
+	redisMaxRetries              = "redisMaxRetries"
+	redisMaxRetryInterval        = "redisMaxRetryInterval"
+	redisMinRetryInterval        = "redisMinRetryInterval"
+	failover                     = "failover"
+	sentinelMasterName           = "sentinelMasterName"
+	redisDB                      = "redisDB"
+	defaultBase                  = 10
+	defaultBitSize               = 0
+	defaultDB                    = 0
+	defaultRedisMaxRetries       = 3
+	defaultRedisMaxRetryInterval = time.Second * 2
+	defaultRedisMinRetryInterval = time.Millisecond * 8
+	defaultEnableTLS             = false
```

Comment:
There are common default among Redis components. Should we define them globally?

Reply:
You must not set global defaults for all fields because different components used different defaults and changing that would be a breaking change. This must be evaluated on a case by case basis.

However, the defaults should be defined in `internal/component/redis` in the metadata parsing code there, there is a switch statement that allows setting defaults specifically for Configuration store, and that's where these defaults must go.